modular_skyrat/modules/cortical_borer/code/cortical_borer.dm

```diff
@@ -0,0 +1,602 @@
+	. += "Chemical Evoltion Points: [chemical_evolution]"
+	. += "Stat Evolution Points: [stat_evolution]"
+	if(host_sugar())
+		. += "Sugar detected! Unable to generate resources!"
+
+/mob/living/simple_animal/cortical_borer/Life(delta_time, times_fired)
+	. = ..()
+
+	if(!inside_human())
+		return
+
+	if(host_sugar())
+		return
+
+	if(chemical_storage < max_chemical_storage)
+		chemical_storage = min(chemical_storage + chemical_regen, max_chemical_storage)
+
+	if(health < maxHealth)
+		health = min(health * health_regen, maxHealth)
+
+	if(timed_maturity < world.time)
+		timed_maturity = world.time + 1 SECONDS
+		maturity_age++
+
+	switch(maturity_age)
+		if(180) //every three minutes, which basically turns into 9 minutes
+			chemical_evolution++
+		if(360)
+			stat_evolution++
+		if(540)
+			maturity_age = 0
+
+//if it doesnt have a mind, let ghosts have it
+/mob/living/simple_animal/cortical_borer/attack_ghost(mob/dead/observer/user)
+	. = ..()
+	if(ckey)
+		return
+	var/choice = tgui_input_list(usr, "Do you want to control [src]?", "Confirmation", list("Yes", "No"))
+	if(choice != "Yes")
+		return
+	to_chat(user, span_warning("As a borer, you have the option to be friendly or not. Note that how you act will determine how a host responds!"))
+	ckey = user.ckey
+	mind.add_antag_datum(/datum/antagonist/cortical_borer)
+
+//inject chemicals into your host
+/datum/action/cooldown/inject_chemical
+	name = "Inject Chemical (10 chemicals)"
+	cooldown_time = 1 SECONDS
+	icon_icon = 'modular_skyrat/modules/cortical_borer/icons/actions.dmi'
+	button_icon_state = "chemical"
+
+/datum/action/cooldown/inject_chemical/Trigger()
+	if(!IsAvailable())
+		to_chat(owner, span_warning("This action is still on cooldown!"))
+		return
+	if(!iscorticalborer(owner))
+		to_chat(owner, span_warning("You must be a cortical borer to use this action!"))
+		return
+	var/mob/living/simple_animal/cortical_borer/cortical_owner = owner
+	if(!cortical_owner.human_host)
+		to_chat(cortical_owner, span_warning("You need a host in order to use this ability!"))
+		return
+	if(cortical_owner.host_sugar())
+		to_chat(cortical_owner, span_warning("Sugar inhibits your abilities to function!"))
+		return
+	if(!cortical_owner.known_chemicals.len)
+		to_chat(cortical_owner, span_warning("You need to learn chemicals first!"))
+		return
+	if(cortical_owner.chemical_storage < 10)
+		to_chat(cortical_owner, span_warning("You require at least 10 chemical units to inject a chemical!"))
+		return
+	cortical_owner.chemical_storage -= 10
+	var/choice = tgui_input_list(cortical_owner, "Choose a chemical to inject!", "Chemical Selection", cortical_owner.known_chemicals)
+	if(!choice)
+		to_chat(cortical_owner, span_warning("No selection made!"))
+		cortical_owner.chemical_storage += 10
+		return
+	cortical_owner.reagent_holder.reagents.add_reagent(choice, 5)
+	cortical_owner.reagent_holder.reagents.trans_to(cortical_owner.human_host, 30, methods = INGEST)
+	StartCooldown()
```

Comment:
```suggestion
	var/choice = tgui_input_list(cortical_owner, "Choose a chemical to inject!", "Chemical Selection", cortical_owner.known_chemicals)
	if(!choice)
		to_chat(cortical_owner, span_warning("No selection made!"))
		return
	cortical_owner.reagent_holder.reagents.add_reagent(choice, 5)
	cortical_owner.reagent_holder.reagents.trans_to(cortical_owner.human_host, 30, methods = INGEST)
	cortical_owner.chemical_storage -= 10
	StartCooldown()
```

Not sure if this is really necessary in any situation tbh but it saves a line

Reply:
necessary in this order to prevent people from cheesing the system